Stanford/JumpstartEngineering/Install/Courses/DisableCourses.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * @file
+ * Enable Courses
+ */
+
+namespace Stanford\JumpstartEngineering\Install\Courses;
+
+/**
+ *
+ */
+class DisableCourses extends \ITasks\AbstractInstallTask {
+
+  /**
+   * Disable Courses module on JSE sites
+   *
+   * @param array $args
+   *   Installation arguments.
+   */
+  public function execute(&$args = array()) {
+
+    module_load_include('module', 'redirect', 'redirect.module');
+    $modules = array( 'stanford_course_views', 'stanford_courses_administration',
+      'stanford_courses', 'stanford_feeds_helper');
+    module_disable($modules);
+
+    if (drupal_uninstall_modules($modules, FALSE)) {
+      drush_log('Disabled and uninstalled modules: ' . implode(', ', $modules), 'ok');
+    } else
+    {
+      drush_log('Error when uninstalling modules: ' . implode(', ', $modules), 'notice');
+    }
+
+    // Publish courses node
+    $query = new \EntityFieldQuery();
+
+    $entities = $query->entityCondition('entity_type', 'node')
+                      ->propertyCondition('type', 'stanford_page')
+                      ->propertyCondition('title', 'Courses Deprecated')
+                      ->execute();
+
+    if (!empty($entities['node'])) {
```

Comment:
I agree with all the examples you provide. I do believe that this case is slightly different from the example above because this function serves multiple tasks. What I would prefer is to have each function serve a single purpose. In this case, I would prefer unpublish/publish to be in it's own function.